### PR TITLE
More app metadata

### PR DESCRIPTION
**Pull request by HaydarAk**
new properties for data apps to store following information:

*ids:DataApp*
- field for Environment Variables
- field to store information about storage requirements of an app

*ids:AppRepresentation*
- runtime environment info (e.g. Java, Docker, ...)
- reference to registry where app can be retrieved

*ids:AppEndpoint
- "UsagePolicy" AppEndpointType
- ids:appEndpointProtocol

### CODE REVIEW

File: codes/AppEndpointType.ttl [hunk 1]
```diff
@@ -32,4 +32,10 @@ idsc:PROCESS_ENDPOINT
     a ids:AppEndpointType;
     rdfs:label "Process endpoint"@en ;
     rdfs:comment "Endpoint is used to start (or control) app processing."@en ;
+.
+
+idsc:USAGE_POLICY_ENDPOINT
+    a ids:AppEndpointType;
+    rdfs:label "usage policy endpoint"@en ;
```
Comment (JohannesLipp) [inline codes/AppEndpointType.ttl:39]: Please capitalize the first word `Usage` to follow the other definitions in this file, thanks

File: examples/APP_RESOURCE.ttl [hunk 2]
```diff
@@ -21,28 +21,26 @@
...
             ids:appEndpoint [
                 a ids:AppEndpoint;
                 ids:appEndpointInformation "I am an app endpoint. I do endpoint things.";
                 ids:appEndpointDocumentation <https://app.swaggerhub.com/apis/app/1337> ;
                 ids:appEndpointType idsc:INPUT_ENDPOINT ;
                 ids:appEndPointPort 5000 ;
                 ids:path "/input";
-                ids:appEndpointMediaType <https://www.iana.org/assignments/media-types/application/json>
+                ids:appEndpointMediaType <https://www.iana.org/assignments/media-types/application/json>;
+                ids:appEndpointProtocol "HTTP/1.1";
```
Comment (JohannesLipp) [inline examples/APP_RESOURCE.ttl:39]: Please evaluate once more, if we should use a string or URI here. The media type in the line just above this one is an ObjectProperty instead. Any choice is okay to me.

**Review by JohannesLipp — CHANGES_REQUESTED** [review]
In general: LGTM!
Small details: Please change the things addressed in my comments at the code, thank you!

File: examples/APP_RESOURCE.ttl (hunk 2 above)
Comment (HaydarAk) [reply examples/APP_RESOURCE.ttl:39]: we are using an URI/IRI here, and are declaring it as a `ids:IANAMediatype` further below (line 47).  Therefore obj. property + IRI should be correct AFAIK.
As you may remember, we have the "abstract" ids:Mediatype superclass with two subclasses `IANA Mediatype` and `Custom Mediatype`. If someone wants to annotate the mediatype, (s)he has to declare it as either a `CustomMediatype` or `IANAMediatype` otherweise it wont validate successfully against our SHACL rules. Therefore it is not legit to just use some IRI.

File: examples/APP_RESOURCE.ttl [hunk 3]
```diff
@@ -21,28 +21,26 @@
     ids:representation [
         a ids:AppRepresentation ;
         ids:mediaType <https://www.iana.org/assignments/media-types/application/zip> ;
-        ids:instance <https://connector.fit.fraunhofer.de/apps/instance/42> ;
+        ids:appDestribtutionService <https://example.com> ;
+        ids:appRunetimeEnvironment "Docker" ;
```
[the remark this answers is not in the document]
Comment (HaydarAk) [reply examples/APP_RESOURCE.ttl:25]: fixed in a234eace2004a25f00da9fce1bcf1e406cfae6eb

File: codes/AppEndpointType.ttl (hunk 1 above)
Comment (HaydarAk) [reply codes/AppEndpointType.ttl]: Fixed in a234eace2004a25f00da9fce1bcf1e406cfae6eb

**Review by JohannesLipp — APPROVED** [review]
Thank you for taking into account my change requests. Good to merge now IMHO, thanks!

**Pull request merged**